akka-javasdk/src/main/scala/akka/javasdk/impl/consumer/ConsumerImpl.scala

```diff
@@ -96,11 +101,22 @@ private[impl] final class ConsumerImpl[C <: Consumer](
 
   private def toSpiEffect(message: Message, effect: Consumer.Effect): Future[Effect] = {
     effect match {
-      case ProduceEffect(msg, metadata) =>
+      case ProduceEffect(msg: Done, metadata) =>
         Future.successful(
           new SpiConsumer.ProduceEffect(
```

Comment:
It looks like we are missing one more spi effect abstraction, except ProduceEffect, ErrorEffect, and IgnoreEffect.
Sth like: `ConsumedEffect` or `ProcessedEffect` to signal that we successfully consumed the message, but we don't want to publish anything.

Reply:
Aha, as in produce for some messages but not all? Would possibly need some more changes in eventing/message publishing in the runtime than just being about the consumer.

Reply:
`Aha, as in produce for some messages but not all?`  - this is done with ProduceEffect + IgnoreEffect, but when I just consume sth without producing, I still have to return either ProduceEffect(Done) or IgnoreEffect, which semantically looks strange. It is hidden from the user-facing API, but visible in the SPI API.